internal/pkg/chain/car.go

```diff
@@ -29,6 +30,13 @@ type carStateReader interface {
 	ChainStateTree(ctx context.Context, c cid.Cid) ([]format.Node, error)
 }
 
+// Fields need to stay lower case to match car's default refmt encoding as
+// refmt can't handle arbitrary casing.
+type carHeader struct {
+	Roots   block.TipSetKey `cbor:"roots"`
+	Version uint64          `cbor:"version"`
+}
```

Comment:
This is a big red flag to me. Duplicating data structures declared in other packages so that we can annotate them is unsustainable. What if such a structure contains a CID? How is this to interact with structures that already define explicit CBOR encoding?

Reply:
After talking online I think we agree that in general the above argument is valid but that it is not a blocker in this special case.

Rehashing some of the above questions for other readers:
> What if such a structure contains a CID

It will need to define its own encoding or use a wrapped cid type because raw cids are unfortunately not enc/decodable into ipld cbor without some kind of wrapper because the go-cid type's binary serialization is one byte off from ipld cbor's assumed binary serialization and go-cid can't know about ipld cbor.

> How is this to interact with structures that already define explicit CBOR encoding?

The answer is that if a type defines `MarshalCBOR() ([]byte, error)` and/or `UnmarshalCBOR([]byte) error` OR `MarshalBinary() ([]byte, error)` and or `UnmarshalBinary()` then our internal library handles automatically.

The right solution is for someone to PR either cbor marshaling functions onto the car header or struct tags like the above so that the type is portable across cbor libraries.  I view this as a nice to have because the duplication is so minor.